Jenkinsfile

```diff
@@ -129,31 +131,68 @@ def getTests(String baseUrl) {
 }
 
 def getScenarios(String baseUrl, String testId){
-    def scenarios = [[Id: "-1", Name: "Default"]]
-    def isScenarioChoosen = "${params['Run with scenarios']}".toBoolean()
-    if (isScenarioChoosen) {
+    def scenarios = [[Id: "0", Name: "Default"]]
+    def isScenarioChoosen = "${params['Run with datasets']}".toBoolean()
+    def isScenarioPresentForTest = scenariosPresent(baseUrl, testId)
+    if (isScenarioChoosen && isScenarioPresentForTest) {
         def jsonSlurper = new JsonSlurper()
         URL url = new URL("${baseUrl}/projects/${params['Project']}/tests/${testId}/scenarios")
-        HttpURLConnection conn = url.openConnection()
-        conn.setRequestMethod("GET")
-        conn.setDoInput(true)
-        conn.setRequestProperty("Authorization", "APIKEY $useMangoApiKey")
-        conn.connect()
-        String content
-        def responseCode = conn.responseCode
-        if (responseCode == 200) {
-            InputStream inputStream = conn.getInputStream()
-            content = inputStream.getText()
-        } else {
-            throw new Exception("Scenarios get request failed with code: ${responseCode}")
-        }
+        String content = getWebResponseContent(url, "Scenarios")
         def scenarioPage = jsonSlurper.parseText(content)
         if (scenarioPage != null) {
             scenarioPage.each { scenario ->
-                echo "${scenario.Name}"        
+                echo "${scenario.Name}"
                 scenarios << [Id: scenario.Id, Name: scenario.Name]
```

Comment:
Do we need this scenario name now since we're not printing it anymore before scenario execution? Does the test provide any information in the log regarding which dataset it's executing as part of the batch execution?

Reply:
since we are not executing each dataset separately, we need not echo it. but we can show it in the results and executing log
![image](https://github.com/infuse-consulting/jenkins/assets/60608318/5b6f4f9e-4e2b-4f24-8410-1213895bc89a)
![image](https://github.com/infuse-consulting/jenkins/assets/60608318/005cac24-6bfa-488e-aa19-ace29fe4aa97)